examples/mouse_to_tile.rs

```diff
@@ -0,0 +1,383 @@
+
+            for (tile_pos, mut tile_label_transform) in tile_label_q.iter_mut() {
+                let tile_center = tile_pos.center_in_world(&grid_size, &map_type);
+                let text_center = tile_center + label_offset;
+                *tile_label_transform = *map_transform;
+                tile_label_transform.translation += text_center.extend(2.0);
+            }
+
+            for window in windows.iter() {
+                for (mut label_text, mut label_transform) in map_type_label_q.iter_mut() {
+                    *label_transform = Transform {
+                        translation: Vec2::new(
+                            -0.5 * window.width() / 2.0,
+                            0.8 * window.height() / 2.0,
+                        )
+                        .extend(0.0),
+                        ..Default::default()
+                    };
+                    *label_text = Text::from_section(
+                        format!("{:?}", map_type.as_ref()),
+                        TextStyle {
+                            font: font_handle.clone(),
+                            font_size: 20.0,
+                            color: Color::BLACK,
+                        },
+                    )
+                    .with_alignment(TextAlignment::CENTER);
+                }
+            }
+        }
+    }
+}
+
+#[derive(Component)]
+struct HighlightedLabel;
+
+pub fn cursor_pos_as_world_pos(
+    window: &Window,
+    camera_q: &Query<(&Transform, &Camera)>,
+) -> Option<Vec2> {
+    window.cursor_position().map(|cursor_pos| {
+        let (cam_t, cam) = camera_q.single();
+        let window_size = Vec2::new(window.width(), window.height());
+
+        // Convert screen position [0..resolution] to ndc [-1..1]
+        let ndc_to_world = cam_t.compute_matrix() * cam.projection_matrix().inverse();
+        let ndc = (Vec2::new(cursor_pos.x, cursor_pos.y) / window_size) * 2.0 - Vec2::ONE;
+        let world_pos = ndc_to_world.project_point3(ndc.extend(-1.0));
+        world_pos.truncate()
+    })
+}
+
+fn highlight_tile_labels(
+    mut commands: Commands,
+    windows: Res<Windows>,
+    tilemap_q: Query<(
+        &TilemapSize,
+        &TilemapGridSize,
+        &TilemapType,
+        &TileStorage,
+        &Transform,
+    )>,
+    highlighted_tiles_q: Query<Entity, With<HighlightedLabel>>,
+    mut tile_label_q: Query<&mut Text, (With<TileLabel>, Without<MapTypeLabel>)>,
+    camera_q: Query<(&Transform, &Camera)>,
+) {
+    for highlighted_tile_entity in highlighted_tiles_q.iter() {
+        if let Ok(mut tile_text) = tile_label_q.get_mut(highlighted_tile_entity) {
+            for mut section in tile_text.sections.iter_mut() {
+                section.style.color = Color::BLACK;
+            }
+            commands
+                .entity(highlighted_tile_entity)
+                .remove::<HighlightedLabel>();
+        }
+    }
+
+    for window in windows.iter() {
```

Comment:
This block of code is probably the most important bit in the entire file for showing people how tile picking works. I think it would be helpful to explain what is happening here with some comments? 

For example:
```rust
// First we grab the mouse world position. This is done by projecting the cursor position into camera space.
if let Some(cursor_world_pos) = cursor_pos_as_world_pos(window, &camera_q) {
// We need to make sure that the cursors world position is correct due to any map transformation.
// This example only shows how to correctly modify the position based on translation
// but additional code would be needed for transforming if the tilemap's scale was changed.
let map_translation = map_transform.translation.truncate();
let world_pos = cursor_world_pos - map_translation;
// Once we have a world position we can transform it into a possible tile position. 
if let Some(tile_pos) = TilePos::from_world_pos(&world_pos, map_size, grid_size, map_type)
```